### ISSUE TITLE
salt-ssh SHIM is broken on python 3

### ISSUE THREAD

**Issue by ezh** [saltstack/salt#40971]
### Description of Issue/Question

before:
```
salt-ssh '*' test.ping
ezh:
    ----------
    retcode:
        1
    stderr:
    stdout:
        Traceback (most recent call last):
          File "<string>", line 155, in <module>
          File "<string>", line 264, in <module>
          File "<string>", line 251, in main
        TypeError: must be str, not bytes
```

after:
```
salt-ssh '*' test.ping
[DEBUG   ] RETCODE localhost: 0
[DEBUG   ] SHIM retcode(0) and command: {
[DEBUG   ] LazyLoaded nested.output
ezh:
    True
```

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

Default configuration.

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

Python 3

```
salt-ssh '*' test.ping
...
[DEBUG   ] Terminal Command: /bin/sh -c ssh localhost -t -t -o KbdInteractiveAuthentication=no -o PasswordAuthentication=no -o GSSAPIAuthentication=no -o ConnectTimeout=300 -o StrictHostKeyChecking=no -o Port=22 -o IdentityFile=/home/ezh/workspace/salt-test/etc/salt/pki/master/ssh/salt-ssh.rsa -o User=ezh  rm '$HOME/.b'3a0a28c4148f'.py'
[DEBUG   ] RETCODE localhost: 1
[DEBUG   ] SHIM retcode(1) and command: Traceback (most recent call last):
[DEBUG   ] LazyLoaded raw.output
{'ezh': {'stdout': 'Traceback (most recent call last):\n  File "<string>", line 155, in <module>\n  File "<string>", line 264, in <module>\n  File "<string>", line 251, in main\nTypeError: must be str, not bytes', 'retcode': 1, 'stderr': 'command-line line 0: Unsupported option "gssapiauthentication"\r\nConnection to localhost closed.\r\n'}}
```

### Versions Report

```
Salt Version:
           Salt: 2016.11.0-1251-gd1ac7b6

Dependency Versions:
           cffi: 1.9.1
       cherrypy: Not Installed
       dateutil: 2.6.0
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.3
          ioflo: Not Installed
         Jinja2: 2.9.5
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.4.5 (default, Jan 28 2017, 19:37:16)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: 4.1.6

System Versions:
           dist: gentoo 2.3
        machine: x86_64
        release: 4.9.9-gentoo
         system: Linux
        version: Gentoo Base System 2.3
```

**Comment by ezh** [comment]
PR will be available shortly

**Comment by cro** [comment]
Thank you for helping us to move toward better Py3 compatibility.

**Comment by gtmanfred** [comment]
Thanks for the pr.

Daniel

**Comment by terminalmage** [comment]
This looks like it can be closed since #40967 was merged.
